templates/dhcpd.conf.j2

```diff
@@ -1,20 +1,41 @@
 {{ ansible_managed | comment }}
-default-lease-time {{ dhcpd_default_lease_time }};
-max-lease-time {{ dhcpd_max_lease_time }};
-option subnet-mask {{ dhcpd_subnet_mask }};
-option broadcast-address {{ dhcpd_broadcast_address }};
-option routers {{ dhcpd_routers }};
-option domain-name-servers {{ dhcpd_domain_name_servers }};
-option domain-search "{{ dhcpd_domain_search }}";
 
-{% for subnet in dhcpd_subnets %}
-subnet {{ subnet.network }} netmask {{ subnet.netmask }} {
-{% if subnet.range_start is defined and subnet.range_end is defined %}
-  range {{ subnet.range_start }} {{ subnet.range_end }};
+# GLOBAL CONFIG SECTION
+{% if 'parameters' in  dhcpd_global_statements %}
+{% for parameter, value in dhcpd_global_statements.parameters.items() %}
+    {{ parameter }} {{ value }};
+{% endfor %}
+{% endif %}
+{% if 'options' in dhcpd_global_statements %}
+{% for option, value in dhcpd_global_statements.options.items() %}
+    option {{ option }} {{ value }};
+{% endfor %}
 {% endif %}
-{% if dhcpd_filename is defined %}  filename "{{ dhcpd_filename }}";
+
+# SUBNET GENERATED SECTION
+{# empoty line is needed for  preaty rendering#}
+
+{% for subnet_statement in dhcpd_subnets %}
+{% if subnet_statement.subnet | ipaddr('network') %}
```

Comment:
Ah, I see where that `netaddr` comes from. Can we do without? That would make it easier to use this role.

I work in an organization where Ansible Tower/AWX is used where I don't have control over the Ansible controller, I can't just install a package I need there...